apps/emqx/src/emqx_session.erl

```diff
@@ -813,7 +813,13 @@ run_terminate_hooks(ClientInfo, takenover, Session) ->
 run_terminate_hooks(ClientInfo, Reason, Session) ->
     run_hook('session.terminated', [ClientInfo, Reason, info(Session)]).
 
-redispatch_shared_messages(#session{inflight = Inflight, mqueue = Q}) ->
+maybe_redispatch_shared_messages(takenover, _Session) ->
+    ?tp(debug, ignore_redispatch_shared_messages, #{reason => takenover}),
+    ok;
+maybe_redispatch_shared_messages(kicked, _Session) ->
```

Comment:
:thinking:  I'm not quite sure why kicked session doesn't need to resend shared-subs messages?

Reply:
The PUBACK for redispatch QoS1 will never get a receiver, turns out to be inconsistent state between client and broker side?